### ISSUE TITLE
190_no_fastdp_2_test hangs on Docker 1.11.0-rc4

### ISSUE THREAD

**Issue by rade** [weaveworks/weave#2141]
Tested with Weave 1.4.6, 1.4 head, and master.

It actually gets to the end, but the process spawned by
```
assert_raises "exec_on $HOST1 c1 sh -c 'nc -l -p 5555 >/tmp/foo'" &
```
hasn't terminated.

This may be a subtle change in behaviour of `docker exec`. May well turn out to be harmless in general, and only affect this test.

**Comment by rade** [comment]
Running against master...

Docker 1.10.3
```
$ weave launch
$ weave run 10.2.1.46/24 --name=c1 -ti alpine /bin/sh
9d4238737f629504a57995ed4e55ba33953b580bbc6b3e70918c28a9d4431b77
$ weave run 10.2.1.47/24 --name=c2 -ti alpine /bin/sh
060c67a3c8a5d57ab64fb110bb29f91028a07cf5bac8416b95ae7d58fa121446
$ docker exec c1 sh -c 'nc -l -p 5555 >/tmp/foo' &
[1] 4702
$ docker exec c2 nc c1 5555
[1]+  Done                    docker exec c1 sh -c 'nc -l -p 5555 >/tmp/foo'
$
```
Docker 1.11.0-rc4
```
$ weave launch
$ weave run 10.2.1.46/24 --name=c1 -ti alpine /bin/sh
a272e31513c05353347916e124c28b0452621401cc34e62311f17815f55d6644
$ weave run 10.2.1.47/24 --name=c2 -ti alpine /bin/sh
b0573d90b66bd613e953dae654cfb07383d9c4e9640804217d578df697cb1d48
$ docker exec c1 sh -c 'nc -l -p 5555 >/tmp/foo' &
[1] 7746
$ docker exec c2 nc c1 5555
```

**Comment by rade** [comment]
btw, I've played with various combinations of `-t` and `-i` here, as well as starting the nc directly in `weave run`. Couldn't find anything that produces the old behaviour.

**Comment by bboreham** [comment]
This command:

    docker exec c1 sh -c 'nc -l -p 5555 >/tmp/foo' &

runs nc in a context where its `stdin` is undefined.  Docker gives it a pipe, the other end of which is held by the docker daemon. Evidently the precise behaviour of this pipe has changed from 1.10 to 1.11 and hence `nc` does not terminate.

If we pin down stdin:

    docker exec c1 sh -c 'nc -l -p 5555 </dev/null >/tmp/foo' &

then `nc` seems happy again

**Comment by bboreham** [comment]
Fixed in 26bd1b57876df27a425d3c62d73dffde3af83ac7
